test/onnx/internal/fx/test_frontend_consistency.py

```diff
@@ -0,0 +1,157 @@
+            device: str,
+            dtype: torch.dtype,
+            module_info: common_modules.ModuleInfo,
+            training: common_modules.TrainEvalMode,
+        ):
+            """Base test method for testing each fx frontend, used by instantiate_device_type_tests."""
+            # device is provided by instantiate_device_type_tests, but we only want to run in cpu.
+            assert device == "cpu"
+
+            module_cls = module_info.module_cls
+            module_inputs = module_info.module_inputs_func(
+                module_info,
+                device=device,
+                dtype=dtype,
+                requires_grad=False,
+                training=False,
+            )
+
+            for (i, module_input) in enumerate(module_inputs):
+                with self.subTest(
+                    module_name=module_cls.__name__,
+                    sample_num=i,
+                    fx_frontend=fx_frontend.name,
+                ):
+                    init_args, init_kwargs = (
+                        module_input.constructor_input.args,
+                        module_input.constructor_input.kwargs,
+                    )
+                    model = module_cls(*init_args, **init_kwargs)
+                    model.to(device).to(dtype)
+                    model.eval()
+
+                    args, kwargs = (
+                        module_input.forward_input.args,
+                        module_input.forward_input.kwargs,
+                    )
+
+                    if isinstance(fx_frontend, frontend.FxFrontendUnpackKwargs):
+                        graph_module, new_args = fx_frontend(model, *args, **kwargs)
+                        fx_outputs = graph_module(*new_args)
+                    else:
+                        graph_module = fx_frontend(model, *args, **kwargs)
+                        fx_outputs = graph_module(*args, **kwargs)
+
+                    outputs = model(*args, **kwargs)
+
+                    self.assertEqual(
+                        pytree.tree_flatten(outputs)[0],
+                        pytree.tree_flatten(fx_outputs)[0],
+                    )
+
+        test_name = f"test_output_match_frontend_{fx_frontend.name}"
+        _output_match_base.__name__ = test_name
+        return _output_match_base
+
+    @classmethod
+    def parameterize_frontends(cls, frontends: Sequence[_FrontendType]):
+        for fx_frontend in frontends:
+            # Generate a test method for each frontend.
+            base_method = cls.create_test_base(fx_frontend)
+            # Important to rename the test method so that DecorateInfo can find it
+            test_name = base_method.__name__
+
+            # TODO(bowbao): Potentially re-use 'add_decorate_info' to have finer-grained
+            # control over which modules to skip or expect to fail, based on frontend.
+
+            setattr(cls, test_name, base_method)
+
+
+TestFxFrontendConsistency.parameterize_frontends(FX_FRONTENDS)
```

Comment:
Why do you need to `parameterize_frontends(....)`? How are the elements in FX_FRONTENDS related to the actual test code? I see you dynamically genearte some `def test_*` in `TestFxFrontendConsistency`. But test code should be easier to read and review. For example,
```
class TestFxFrontendConsistency(common_utils.TestCase):
  def test_first_frontend(...):
  def test_second_frontend(...)
```

Reply:
Commented above.